cmd/virt-launcher/virt-launcher.go

```diff
@@ -312,28 +312,19 @@ func writeProtectPrivateDir(uid string) {
 	}
 }
 
-func cleanupEphemeralDiskDirectory(ephemeralDiskDir string) {
+func cleanupContainerDiskDirectory(ephemeralDiskDir string) {
 	// Cleanup the content of ephemeralDiskDir, to make sure that all containerDisk containers terminate
-	if _, err := os.Stat(ephemeralDiskDir); !os.IsNotExist(err) {
-		dir, err := ioutil.ReadDir(ephemeralDiskDir)
-		if err != nil {
-			log.Log.Reason(err).Errorf("failed to read content of the ephemeral disk directory: %s", ephemeralDiskDir)
-			return
-		}
-		for _, d := range dir {
-			removePath := filepath.Join(ephemeralDiskDir, d.Name())
-			err := os.RemoveAll(removePath)
-			if err != nil {
-				log.Log.Reason(err).Errorf("could not clean up ephemeral disk directory: %s", removePath)
-			}
-		}
+	err := RemoveContents(ephemeralDiskDir)
+	if err != nil {
+		log.Log.Reason(err).Errorf("could not clean up ephemeral disk directory: %s", ephemeralDiskDir)
 	}
 }
 
 func main() {
 	qemuTimeout := pflag.Duration("qemu-timeout", defaultStartTimeout, "Amount of time to wait for qemu")
 	virtShareDir := pflag.String("kubevirt-share-dir", "/var/run/kubevirt", "Shared directory between virt-handler and virt-launcher")
 	ephemeralDiskDir := pflag.String("ephemeral-disk-dir", "/var/run/kubevirt-ephemeral-disks", "Base directory for ephemeral disk data")
+	containerDiskDir := pflag.String("container-disk-dir", "/var/run/kubevirt/container-disks", "Base directory for container disk data")
```

Comment:
does this mean that every vmi's pod can access every other vmi's containerDisks now?

Reply:
No, the containerdisks are directly mounted into the correct pod at these paths. If you go to /var/run/kubevirt/container-disks on the node, you will just see an empty directory.

Reply:
yay, alrighty. that's what I thought, i just second guessed myself. volumes involve so much indirection at times